### DISCUSSION TITLE
Automatically generate the pages and subdirectories from backend data or Strapi

### DISCUSSION THREAD

**Discussion by asdutoit** [shuding/nextra#275] [Ideas]
Would / Is it possible to have Nextra generate the pages and categories by pulling data from a separate backend service, such as a headless CMS (Strapi fro example)? 

We utilise Strapi for various projects from a back-end perspective.    We can for example, write our content in markdown on strapi which can be saved to a db.  Nextra can then on page refresh or (getStaticProps) pull the data from the Strapi api and build the pages.

**Comment by The-Code-Monkey** [comment]
Yeah I was thinking of doing this, did you ever get anywhere with it. In terms of just requesting the data from an API source that would return the markdown??

**Comment by atowersc** [comment]
Was this achieved? 👀
